### PR TITLE
Fix to check if the repo is enabled and don’t make a call to the sett…

### PR DESCRIPTION
**Pull request by josmo**
@bradrydzewski if you don't mind taking a look and merging as soon as you can.  I didn't realize that the hook worked different the first time you ever enable the repo vs once it's ever been enabled even if it's off :/  Right now you can't enable brand new repos but this fixes it.

### CODE REVIEW

File: remote/bitbucketserver/internal/types.go [hunk 1]
```diff
@@ -170,6 +170,19 @@ type RefChange struct {
...
+			Name string `json:"name"`
+			Type string `json:"type"`
+			Description string `json:"description"`
+			Version string `json:"version"`
+			ConfigFormKey string `json:"configFormKey"`
+		} `json:"details"`
+	Enabled bool `json:"enabled"`
+	Configured bool `json:"configured"`
+}
+
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/types.go]: needs gofmt

File: remote/bitbucketserver/internal/client.go [hunk 2]
```diff
@@ -131,15 +132,23 @@ func (c *Client) FindFileForRepo(owner string, repo string, fileName string, ref
 }
 
 func (c *Client) CreateHook(owner string, name string, callBackLink string) error {
-	hookSettings, err := c.GetHooks(owner, name)
+	hookDetails, err := c.GetHookDetails(owner, name)
 	if err != nil {
 		return err
 	}
-	hooks := hookSettingsToArray(hookSettings)
+	hooks := make([]string, 0)
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: you can declare like this
```
var hooks []string
```

File: remote/bitbucketserver/internal/client.go [hunk 3]
```diff
@@ -165,6 +174,22 @@ func (c *Client) DeleteHook(owner string, name string, link string) error {
...
 }
 
+func (c *Client) GetHookDetails(owner string, name string) (*HookPluginDetails, error) {
+	urlString := fmt.Sprintf(pathHookDetails, c.base, owner, name, hookName)
+	response, err := c.client.Get(urlString)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: should handle this error

Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: also you can combine reading the body and unmarshaling into a single command, which will be a bit more efficient:

```
json.NewDecoder(response.Body).Decode(&hookDetails)
```

File: remote/bitbucketserver/internal/client.go [hunk 4]
```diff
@@ -165,6 +174,22 @@ func (c *Client) DeleteHook(owner string, name string, link string) error {
...
+	hookDetails := HookPluginDetails{}
+	err = json.Unmarshal(contents, &hookDetails)
+	if err != nil {
+		return nil, err
+	}
+	return &hookDetails, nil
```
Comment (bradrydzewski) [inline remote/bitbucketserver/internal/client.go]: in this case I would `return &hookDetails, err`. The calling function will generally check` err != nil` so there is no harm returning both

**Comment by bradrydzewski** [comment]
sorry had a few more minor comments :)

**Comment by josmo** [comment]
@bradrydzewski no worries and thanks for the comments.  Hopefully I'll get more consistent with my PRs :)

**Pull request merged**